app/components/GenericModal.tsx

```diff
@@ -118,6 +122,10 @@ const GenericModal = (
     window.location.hash = '#';
   };
 
+  if (typeof children === 'function') {
```

Comment:
What is this doing? Is there a reason why the children can be both a function and an object?

Reply:
there is a usecase to pass function as the children to access to the context information;
https://github.com/bcgov/sso-requests/blob/49c6e83a819b356e8f3f3c914046d5e8853e2e9b/app/page-partials/my-dashboard/UserRoles.tsx#L499

Reply:
Cool thanks for clearing up :)